### ISSUE TITLE
[Bug]: Focus Bottom Window hotkey focuses wrong window

### ISSUE THREAD

**Issue by jsjuris89** [Bismuth-Forge/bismuth#221]
### Summary

`Meta + K` or Focus Bottom Window doesn't go to correct window, rest `Focus` hotkeys work as expected.

### Steps to Reproduce

1. Tile Layout 2
2. 4 windows open
3. Try go from top window to bottom window
4. Goes wrong (see screenshot)

### Expected behavior

Should go to bottom window.

### Screenshots

![](https://i.imgur.com/jFNrcW4.png)
Pressing `Meta + K` on 1st window goes to 4th (should go 2nd) window.
Pressing `Meta + K` on 3rd window goes to 2nd (should go 4th) window

### Bismuth version

AUR - kwin-bismuth 2.2.0-1

### KDE Plasma version

5.23.4

### The platform KWin is running on

X11

**Comment by gikari** [comment]
Cannot reproduce, but It is possible, that this behavior could break due to #209 or #208.
If anyone else could reproduce and/or make more precise instructions on how to reproduce (there might be a difference in our configurations, that prevents reproducing), I will take a look.

**Comment by gikari** [comment]
@jsjuris89 Could you reproduce your issue with #222 applied?

**Comment by jsjuris89** [comment]
@gikari I could try. What I need to do?

**Comment by gikari** [comment]
Since it was just merged, you need to:

1. Remove AUR version.
2. Run these (make sure you have NPM installed):
    ```bash
    git clone https://github.com/Bismuth-Forge/bismuth.git
    cd bismuth
    npm install # Installs dependencies for building
    npm run sysdep-install # Install system dependencies for building
    npm run bi-install # Installs all Bismuth components
    ```
3. Try to reproduce the bug.

You may want to roll back to the AUR version after that, to do that, uninstall the Bismuth with:
```bash
npm run bi-uninstall
```

**Comment by jsjuris89** [comment]
@gikari I got this error:
```
[elnath@elnath-ryzen bismuth]$ npm run sysdep-install

> bismuth@2.2.0 sysdep-install
> scripts/sysdep-install.sh

⛓️ Installing system dependencies...
⚠ Your distribution is EndeavourOS, but you have to install system dependencies manually.
[elnath@elnath-ryzen bismuth]$
```

**Comment by gikari** [comment]
> @gikari I got this error:
> 
> ```
> [elnath@elnath-ryzen bismuth]$ npm run sysdep-install
> 
> > bismuth@2.2.0 sysdep-install
> > scripts/sysdep-install.sh
> 
> ⛓️ Installing system dependencies...
> ⚠ Your distribution is EndeavourOS, but you have to install system dependencies manually.
> [elnath@elnath-ryzen bismuth]$
> ```

You can open an issue about it. For now, just execute this:
```
      sudo pacman -S --noconfirm --needed \
        cmake ninja extra-cmake-modules
```

**Comment by jsjuris89** [comment]
@gikari Ok I did that. Still have same bug.

**Comment by gikari** [comment]
I forgot to mention, that you need to restart kwin with:
```
kwin_x11 --replace & disown
```
If you didn't do that, the Bismuth script is not up-to-date.
You can of course log in and log out to do the same.

**Comment by jsjuris89** [comment]
@gikari I tried both log out and the command above. Still same behaviour. Only `Focus bottom window` is bugged.
Here is additional info:
![](https://i.imgur.com/uuLkJIr.png)
If I keep pressing `Meta + K` it goes from 1 -> 4 -> 3 -> 2 -> 1

**Comment by gikari** [comment]
This is really weird. It's very similar to the behavior of the "Previous Window"  shortcut. Are you sure Meta K is bound correctly?

You can check it by going to "System Setting" > "Shortcuts" > "KWin" and filtering the results by "Bismuth" keyword.

**Comment by jsjuris89** [comment]
@gikari Alright, we getting closer, this is what I found in hotkeys:
![](https://i.imgur.com/ekoRGbF.png)
So I reassigned `Meta K` to Bottom Window.

The problem is that hotkey is not working. Pressing `Meta + K` does nothing.

**Comment by gikari** [comment]
KWin shortcuts can be buggy when reassigning from old shortcuts. Close and reopen the same shortcut dialog and check if the shortcut is bound.

**Comment by jsjuris89** [comment]
Yes, I cleared `Focus bottom window` to no hotkey, hit applied. And tried again `Meta + K` now it works. So the problem was not in the `bismuth` but there are some bugs in `System Setting` > `Shurtcuts` ? @gikari

**Comment by gikari** [comment]
I don't know, lol. The rebinding from one shortcut to another was random and from what I checked, there isn't anything, that could cause that in the Bismuth code base.

Could you please report the non-applying-rebinding bug to upstream kwin and link it here? You need to go to https://bugs.kde.org to do that.

> KWin shortcuts can be buggy when reassigning from old shortcuts. Close and reopen the same shortcut dialog and check if the shortcut is bound.

**Comment by jsjuris89** [comment]
> Could you please report the non-applying-rebinding bug to upstream kwin and link it here?

Sure I can try, but tomorrow sleep time now.
